app/controllers/api/alls_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Api::AllsController < ApplicationController
+  def show
+    contest_slug = params[:contest_slug]
+    query = "SELECT submits.*, problems.id, problems.contest_id, contests.id, contests.slug
+    FROM (submits INNER JOIN problems ON submits.problem_id = problems.id) 
+    INNER JOIN contests ON problems.contest_id = contests.id
+    WHERE contests.slug = :contest_slug;"
```

Comment:
SQL直書きではなくてActiveRecordのクエリで処理してください

```suggestion
    submits.joins(problem: :contest).select('submits.*, problems.id, problems.contest_id, contests.id, contests.slug).where('contests.slug = ?', contest_slug)
```